MediQuiz/AppDelegate.swift

```diff
@@ -10,12 +10,74 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
         // Override point for customization after application launch.
 
         window = UIWindow(frame: UIScreen.main.bounds)
-        let viewController = ViewController()
+        let viewController = CategoryViewController()
         window?.rootViewController = UINavigationController(rootViewController: viewController)
         window?.makeKeyAndVisible()
         
+        isAppAlreadyLaunchedOnce()
+        
         return true
     }
+    
+    private func isAppAlreadyLaunchedOnce() -> Bool {
+        let defaults = UserDefaults.standard
+        if defaults.bool(forKey: "isAppAlreadyLaunchedOnce") && defaults.integer(forKey: "questionCount") == countDataFromJson() {
+            return true
+        } else {
+            defaults.set(true, forKey: "isAppAlreadyLaunchedOnce")
+            deleteData()
+            loadDataFromJSON()
+        }
+        
+        return false
+    }
+    
+    private func countDataFromJson() -> Int {
+        DataLoader().questionModel.count
+    }
```

Comment:
to lepiej zrobić jako zmienną i jeszcze lepiej to umieścić ją w DataLoader i stamtąd ją wyciągać:

```suggestion
    private var questionsCount: Int {
        questionModel.count
    }
```